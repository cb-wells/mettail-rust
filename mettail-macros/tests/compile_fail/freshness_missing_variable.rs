use mettail_macros::theory;

theory! {
    name: Invalid,
    
    exports {
        Name
    }
    
    terms {
        NZero . Name ::= "@0" ;
    }
    
    equations {
        // Error: freshness variable 'x' doesn't appear in equation
        if x # Q then (NZero) == (NZero)
    }
}

fn main() {}

