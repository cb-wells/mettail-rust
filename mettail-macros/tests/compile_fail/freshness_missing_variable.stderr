error: expected `;`
  --> tests/compile_fail/freshness_missing_variable.rs:17:5
   |
17 |     }
   |     ^
